tests/product_test.py

```diff
@@ -40,4 +36,19 @@ def test_product_response_self_link(
         warn(StapiSpecWarning("GET /products Link[rel=self] should exist"))
     else:
         assert link["type"] == "application/json"
-        assert link["href"] == url_for("/products/mock:standard")
+        assert link["href"] == url_for(f"/products/{product_id}")
+
+
+@pytest.mark.parametrize("product_id", ["test-spotlight"])
+def test_product_constraints_response(
+    product_id: str,
+    stapi_client: TestClient,
+):
+    res = stapi_client.get(f"/products/{product_id}/constraints")
+    assert res.status_code == status.HTTP_200_OK
+    assert res.headers["Content-Type"] == "application/json"
+
+    data = res.json()
```

Comment:
Get more eyes on this to make sure this is the correct format

Reply:
This is the shape of the whole response from `res.json()`

```json
{
    "$defs": {
        "DatetimeInterval": {
            "maxItems": 2,
            "minItems": 2,
            "prefixItems": [
                {
                    "format": "date-time",
                    "type": "string"
                },
                {
                    "format": "date-time",
                    "type": "string"
                }
            ],
            "type": "array"
        }
    },
    "additionalProperties": true,
    "properties": {
        "datetime": {
            "$ref": "#/$defs/DatetimeInterval"
        },
        "off_nadir": {
            "title": "Off Nadir",
            "type": "integer"
        }
    },
    "required": [
        "datetime",
        "off_nadir"
    ],
    "title": "TestSpotlightProperties",
    "type": "object"
}
```